### ISSUE TITLE
Make recorder standalone

### ISSUE THREAD

**Issue by smiklosovic** [arquillian/arquillian-recorder#27]
Currently, recorder listens to classsic Before/After Class/Test events and acts respectively.

This should be externalized and core event logic should be agnostic about who these events fired.

Right now it is done like this:

BeforeTest -> @BeforeTest -> reporting

In order to abstract this, it should be

BeforeTest -> @BeforeTest -> BeforeTestReportEvent -> @Observes BeforeTestReportEvent -> reporting

So some standalone implementation can do

BeforeTestReportEvent -> @Observes BeforeTestReportEvent -> reporting

**Comment by smiklosovic** [comment]
I am not sure yet if this is ever neccessary, because I could reuse what is already implemented and fire Before and After events on my own but I am not sure if that triggers any other core observers from Arquillian.

**Comment by aslakknutsen** [comment]
Absolutely. Extensions should generally separate between it's own action command and integration commands.
